scene/gui/tree.cpp

```diff
@@ -4215,10 +4215,22 @@ Tree::SelectMode Tree::get_select_mode() const {
 }
 
 void Tree::deselect_all() {
+	if (root) {
+		for (int i = 0; i < columns.size(); ++i) {
+			if (root->is_selected(i)) {
+				root->deselect(i);
+			}
+		}
+	}
+
 	TreeItem *item = get_next_selected(get_root());
 	while (item) {
-		for (int i = 0; i < columns.size(); i++) {
-			item->deselect(i);
+		if (select_mode == SELECT_ROW) {
+			item->deselect(0);
```

Comment:
Similar question here: is this optimization actually useful? If it is, why don't we do it for the root node?